dnet/dnet_xdag.c

```diff
@@ -355,46 +430,69 @@ static void *xthread_main(void *arg)
 					size = SECTOR_SIZE;
 				}
 				res = read(t->fds[n].fd, xs->byte + SECTOR_SIZE - size, size);
-				if(res <= 0) { err = res << 4 | 3; goto close; }
+				if(res <= 0) {
+					err = res << 4 | 3;
+					goto close;
+				}
+
 				if(res != size) {
 					if(!conn->part) {
 						conn->part = malloc(sizeof(struct xpartbuf));
-						if(!conn->part) { err = 4; goto close; }
+						if(!conn->part) {
+							err = 4;
+							goto close;
+						}
 						conn->part->writelen = 0;
 					}
-					if(xs != &conn->part->read)
+					if(xs != &conn->part->read) {
 						memcpy(&conn->part->read, xs, res);
+					}
 					conn->part->readlen = size - res;
 					goto skip_in;
-				} else if(conn->part)
+				} else if(conn->part) {
 					conn->part->readlen = 0;
+				}
+
 				if(conn->packets_in >= FIRST_NSECTORS) {
 					dfslib_uncrypt_sector((conn->crypt ? &conn->crypt->crypt_in : g_crypt),
 						xs->word, conn->packets_in - FIRST_NSECTORS + 1);
 					ttl = xs->head.ttl;
 					if(xs->head.type != DNET_PKT_XDAG || xs->head.length != SECTOR_SIZE || !ttl) {
-						conn->dropped_in++; goto decline;
+						conn->dropped_in++;
+						goto decline;
 					}
 					uint32_t crc = xs->head.crc32;
 					xs->head.crc32 = 0;
-					if(crc_of_array(xs->byte, SECTOR_SIZE) != crc) { conn->dropped_in++; goto decline; }
-					if(!g_arrive_callback) goto decline;
+					if(crc_of_array(xs->byte, SECTOR_SIZE) != crc) {
+						conn->dropped_in++;
+						goto decline;
+					}
+					if(!g_arrive_callback) {
+						goto decline;
+					}
 					res = (*g_arrive_callback)(xs->word, conn);
-					if(res < 0) goto decline;
+					if(res < 0) {
+						goto decline;
+					}
 					if(res > 0 && ttl > 2) {
 						xs->head.ttl = ttl;
 						dnet_send_xdag_packet(xs->word, (void *)((uintptr_t)conn | 1));
 					}
 				} else {
 					if(!conn->crypt) {
 						if(memcmp((struct xsector *)&g_xkeys.pub + conn->packets_in, xs, SECTOR_SIZE)) {
-							if(conn->packets_in) { err = conn->packets_in << 4 | 5; goto close; }
+							if(conn->packets_in) {
+								err = conn->packets_in << 4 | 5;
+								goto close;
+							}
 							conn->crypt = (struct xcrypt *)malloc(sizeof(struct xcrypt));
-							if(!conn->crypt) { err = 6; goto close; }
+							if(!conn->crypt) {
+								err = 6;
+								goto close;
+							}
 							conn->crypt->last_sent = time(0);
 						}
-					}
-					if(conn->crypt) {
+					} else {
```

Comment:
conn-crypt is modified
`							conn->crypt = (struct xcrypt *)malloc(sizeof(struct xcrypt));
`
that's why there is 					`if(conn->crypt) {
`
and not` } else {`

Reply:
Ah, @sgaragagghu I did't see your comments. Yes, it is a good point

Reply:
Thank you @sgaragagghu